src/lambda/commands/deploySamApplication.ts

```diff
@@ -72,34 +94,42 @@ export async function deploySamApplication({
                 stackName,
                 region,
                 parameterOverrides,
-                invoker
+                invoker,
+                profile
             )
             // Deploying can take a very long time for Python Lambda's with native dependencies so user needs feedback
             restParams.outputChannel.appendLine(localize(
-                'AWS.samcli.deploy.stackName.initiated',
-                'Deploying {0} stack...',
+                'AWS.samcli.deploy.workflow.stackName.initiated',
+                'Deploying SAM Application to CloudFormation Stack: {0}',
                 stackName
             ))
             await deployInvocation.execute()
-            // TODO: Add nls support
-            const msg = `Successfully deployed SAM Application to CloudFormation Stack: ${stackName}`
-            restParams.outputChannel.appendLine(msg)
-            // TODO: Is this the right way to provide this feedback?
-            vscode.window.showInformationMessage(msg)
+
+            msg = localize(
+                'AWS.samcli.deploy.workflow.success',
+                'Successfully deployed SAM Application to CloudFormation Stack: {0} with profile: {1}',
+                stackName, profile
+            )
         } catch (err) {
-            // TODO: Add nls support
-            let msg = `Failed to deploy SAM application. Error while ${stage}: ${String(err)}`
+            msg = localize(
+                'AWS.samcli.deploy.workflow.error',
+                'Failed to deploy SAM application. Error while {0}: {1}',
+                stage, String(err)
+            )
             // tslint:disable-next-line:max-line-length
             // detect error message from https://github.com/aws/aws-cli/blob/4ff0cbacbac69a21d4dd701921fe0759cf7852ed/awscli/customizations/cloudformation/exceptions.py#L42
             // and append region to assist in troubleshooting the error
             // (command uses CLI configured value--users that don't know this and omit region won't see error)
             if (msg.includes(`aws cloudformation describe-stack-events --stack-name ${args.stackName}`)) {
                 msg += ` --region ${args.region}`
+                if (profile) {
+                    msg += ` --profile ${profile}`
+                }
             }
+        } finally {
```

Comment:
This "trailing msg handling" is prone to misunderstanding in future code maintenance. Remove the method-wide msg variable and handle it explicitly in each place (line 112 and 128)